tests/ops/functions/test_bind_new_parameters.py

```diff
@@ -215,6 +226,45 @@ def test_hamiltonian(H, new_coeffs, expected_H):
     assert new_H is not H
 
 
+TEST_BIND_LINEARCOMBINATION = [
+    (
+        qml.ops.LinearCombination(
+            [1.1, 2.1, 3.1],
+            [qml.prod(qml.PauliZ(0), qml.PauliX(1)), qml.Hadamard(1), qml.PauliY(0)],
+        ),
+        [1.2, 2.2, 3.2],
+        qml.ops.LinearCombination(
+            [1.2, 2.2, 3.2],
+            [qml.prod(qml.PauliZ(0), qml.PauliX(1)), qml.Hadamard(1), qml.PauliY(0)],
+        ),
+    ),
+    (
+        qml.ops.LinearCombination(
+            [1.6, -1], [qml.Hermitian(np.array([[0.0, 1.0], [1.0, 0.0]]), wires=1), qml.PauliX(1)]
+        ),
+        [-1, np.array([[1.0, 1.0], [1.0, 1.0]]), 1.6],
+        qml.ops.LinearCombination(
+            [-1, 1.6], [qml.Hermitian(np.array([[1.0, 1.0], [1.0, 1.0]]), wires=1), qml.PauliX(1)]
+        ),
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "H, new_coeffs, expected_H",
+    TEST_BIND_LINEARCOMBINATION,
+)
+def test_linear_combination(H, new_coeffs, expected_H):
+    """Test that `bind_new_parameters` with `LinearCombination` returns a new
+    operator with the new parameters without mutating the original
+    operator."""
+    new_H = bind_new_parameters(H, new_coeffs)
+
+    assert qml.equal(new_H, expected_H)
+    assert new_H is not H
```

Comment:
Can you also check that the `operands` and `ops` are not the same as the original operator's `operands` and `ops`?

Reply:
```suggestion
    assert new_H is not H
    assert all(no is not o for no, o in zip(new_op.operands, op.operands))
    assert all(no is not o for no, o in zip(new_op.ops, op.ops))
```

Reply:
> `assert all(no is not o for no, o in zip(new_op.ops, op.ops))`

For the latter I would have to make a copy of the `ops` in `bind_new_paramters`, but I actually dont think that is desired/necessary?

Reply:
Ah, I apologize, I forgot that we were not copying non-parametric ops. Feel free to revert the deepcopying change you made and disregard this comment about testing.

Reply:
okay thanks! :)